chainstate/src/chainstate_interface.rs

```diff
@@ -29,17 +29,17 @@ pub trait ChainstateInterface: Send {
     fn subscribe_to_events(&mut self, handler: Arc<dyn Fn(ChainstateEvent) + Send + Sync>);
     fn process_block(&mut self, block: Block, source: BlockSource) -> Result<(), ChainstateError>;
     fn preliminary_block_check(&self, block: Block) -> Result<Block, ChainstateError>;
-    fn get_best_block_id(&self) -> Result<Id<Block>, ChainstateError>;
+    fn get_best_block_id(&self) -> Result<Id<GenBlock>, ChainstateError>;
     fn is_block_in_main_chain(&self, block_id: &Id<Block>) -> Result<bool, ChainstateError>;
     fn get_block_height_in_main_chain(
         &self,
-        block_id: &Id<Block>,
+        block_id: &Id<GenBlock>,
     ) -> Result<Option<BlockHeight>, ChainstateError>;
     fn get_best_block_height(&self) -> Result<BlockHeight, ChainstateError>;
     fn get_block_id_from_height(
         &self,
         height: &BlockHeight,
-    ) -> Result<Option<Id<Block>>, ChainstateError>;
+    ) -> Result<Option<Id<GenBlock>>, ChainstateError>;
```

Comment:
I kept the chainstate interface changes rather minimal, but this could be extended to eliminate some repeating patterns (e.g. match on block id to figure out if it's genesis, call something if it's not, etc)